modular_skyrat/modules/container_emotes/code/container_emotes.dm

```diff
@@ -0,0 +1,67 @@
+/mob/living/verb/container_emote()
+	set name = "Emote Using Vehicle/Container"
+	set category = "IC"
+
+	if (isturf(src.loc))
+		to_chat(src, span_danger("You are not within anything!"))
+		return
+	if (loc && (!src.IsUnconscious()))
+		usr.emote("exme")
+
+/datum/emote/container_emote
+	key = "exme"
+	key_third_person = "exme"
+	message = null
+
+/datum/emote/container_emote/run_emote(mob/living/user, params, type_override = null, intentional = TRUE)
+	var/container_message
+	var/container_emote = params
+	if(QDELETED(user))
+		return FALSE
+	if(is_banned_from(user, "emote"))
+		to_chat(user, "You cannot send emotes (banned).")
+		return FALSE
+	else if(user.client?.prefs?.muted & MUTE_IC)
+		to_chat(user, "You cannot send IC messages (muted).")
+		return FALSE
+	else if (isturf(user.loc))
+		to_chat(user, "You are not within anything!")
+		return FALSE
+	else if(!params)
+		container_emote = tgui_input_text(user, "What would you like to emote?", "Container Emote" , null, MAX_MESSAGE_LEN, TRUE, TRUE, 0)
+		if(!container_emote)
+			return FALSE
+		var/list/choices = list("Visible","Audible")
+		var/type = tgui_input_list(user, "Is this a visible or audible emote?", "Container Emote", choices, 0)
+		switch(type)
+			if("Visible")
+				emote_type = EMOTE_VISIBLE
+			if("Audible")
+				emote_type = EMOTE_AUDIBLE
+			else
+				alert("Unable to use this emote, must be either audible or visible.")
```

Comment:
I think there's TGUI alerts, isn't there?

Reply:
Maybe, I'll check

Reply:
You know what, fuck it, I'll make it a bubble alert, we love bubble alerts here :)